narwhals/_arrow/group_by.py

```diff
@@ -37,12 +37,15 @@ def get_function_name_option(function_name: str) -> Any | None:
 
 
 class ArrowGroupBy:
-    def __init__(self, df: ArrowDataFrame, keys: list[str]) -> None:
+    def __init__(
+        self, df: ArrowDataFrame, keys: list[str], *, drop_null_keys: bool
+    ) -> None:
         import pyarrow as pa  # ignore-banned-import()
 
         self._df = df
         self._keys = list(keys)
         self._grouped = pa.TableGroupBy(self._df._native_frame, list(self._keys))
+        self._drop_null_keys = drop_null_keys
```

Comment:
🤔 maybe this could just be
```
if self._drop_null_keys:
    self._grouped = pa.TableGroupBy(self._df.drop_nulls(self._keys)._native_frame, list(self._keys))
else:
    self._grouped = pa.TableGroupBy(self._df._native_frame, list(self._keys))
```
, same for Polars